classes/battle/BattleEffectsManager.php

```diff
@@ -211,10 +211,10 @@ public function applyPassiveEffect(Fighter $target, BattleEffect $effect): bool
             $target->genjutsu_boost += $effect->effect_amount;
         }
         else if($effect->effect == 'cast_speed_boost') {
-            $target->cast_speed_boost += $target->cast_speed * ($effect->effect_amount / 100);
+            $target->cast_speed_boost += ($target->cast_speed + $target->bloodline_cast_speed_boost) * ($effect->effect_amount / 100);
```

Comment:
To make this slightly more structured, could we extract this to a `getCastSpeed(bool $include_bloodline = false)` method in the Fighter class, and call it here as `$target->getCastSpeed(true) * ($effect->effect_amount / 100)`?

Reply:
Agreed, I made the revisions.